community/bolt/src/main/java/org/neo4j/bolt/v1/runtime/internal/SessionStateMachine.java

```diff
@@ -311,6 +313,31 @@ protected State onNoImplementation( SessionStateMachine ctx, String command )
                         return ERROR;
                     }
                 },
+        /** Failed to authenticate, acknowledging the error here leads you back to UNINITIALIZED and you get to try again */
+        AUTH_FAILURE
+                {
+                    @Override
+                    public State reset( SessionStateMachine ctx )
+                    {
+                        // There may still be a transaction open, so do
+                        // an extra reset on the outcome of ackFailure to ensure we go
+                        // to idle state.
```

Comment:
As I pointed in https://github.com/neo4j/neo4j/issues/7356. I think that line should be simple return `UNINITIALIZED `